fuel-vm/src/verification.rs

```diff
@@ -0,0 +1,64 @@
+//! Stategies for verifying the correctness of the VM execution.
+//! The default strategy, [`Panic`], simply panics on failed verification.
+
+use alloc::collections::BTreeSet;
+
+use fuel_tx::{
+    ContractId,
+    PanicReason,
+};
+
+use crate::{
+    error::PanicOrBug,
+    interpreter::PanicContext,
+    storage::InterpreterStorage,
+};
+
+/// Do not allow outside implementations for the Verifier, so that it's not a breaking
+/// change to modify it.
+trait Seal {}
+
+/// What to do when verification fails.
+#[allow(private_bounds)] // For selaed trait
+pub trait Verifier<S>
+where
+    Self: Sized + Seal,
+    S: InterpreterStorage,
```

Comment:
Do we need `S` here at all? Based on the use cases that we discussed, it looks like we don't need it anymore=)